src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -58,7 +58,7 @@ public void AddAllTo(HashSet<Checksum> checksums)
         AssetPath assetPath,
         TextDocumentStates<TState> documentStates,
         HashSet<Checksum> searchingChecksumsLeft,
-        Dictionary<Checksum, object> result,
+        Action<Checksum, object> onAssetFound,
```

Comment:
Now instead of populating a whole dictionary (and only sending the data once the search finishes), we now just callback into this action.  That action puts hte data into a channel that our pipe-writing code is pulling from.  So we can write out the data the instance we find it.